src/video_core/shader/shader_ir.h

```diff
@@ -186,7 +218,16 @@ class ShaderIR final {
         return num_custom_variables;
     }
 
+    std::shared_ptr<ShaderFunctionIR> GetMainFunction() const {
```

Comment:
```suggestion
    std::shared_ptr<ShaderFunctionIR>& GetMainFunction() const {
```

This should also be nodiscard, ditto for the const variant.